File: Solucion de Sistemas de Ecuaciones/Metodos Directos/Metodo de Thomas/tridiagonal.py
import numpy as np

def tridiagonal(a,b,c,n):
    out=np.zeros((n,n))
    for i in range(n):
        if(i==0):
            out[i,i+1]=c
        elif(i==n-1):
            out[i,i-1]=a
        else:
            out[i,i+1]=c
            out[i,i-1]=a
        out[i,i]=b
    return out

File: Solucion de Sistemas de Ecuaciones/Metodos Directos/Metodo de Thomas/Tridiagonal.m
function test
  tridiagonal(1,2,3,4)
end
function[a_mat]=tridiagonal(a,b,c,n)
  clc;
  a_mat=zeros(n,n);
  for i=1:n
    if i==1
      a_mat(i,i+1)=b;
    elseif i==n-1
      a_mat(i,i-1)=c;
    else
      a_mat(i,i+1)=b;
      a_mat(i,i-1)=c;
    endif
    a_mat(i,i)=a;
  endfor
  return
end